app/models/merchant.rb

```diff
@@ -9,4 +9,21 @@ def self.random
     Merchant.find(id)
   end
 
+  def items
+    Item.where(merchant_id: id)
+  end
+
+  def pending_customers
+    Customer.joins(invoices: :transactions).where('invoices.merchant_id = ?', id).pluck('transactions.result')
+    binding.pry
+  end
+
+  def favorite_customer
```

Comment:
same here - this logic looks solid, but make sure you test this with a unit test!